### PR TITLE
ES6: Migrate 'view_model'

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

File: app/script/view_model/AuthViewModel.js [hunk 1]
```diff
@@ -0,0 +1,1993 @@
...
+      },
+      FORWARDED_URL_PARAMETERS: [
+        z.auth.URLParameter.BOT,
+        z.auth.URLParameter.CALLING_V3,
+        z.auth.URLParameter.ENVIRONMENT,
+        z.auth.URLParameter.LOCALYTICS,
+      ],
+    };
+
+  }
```
Comment (ffflorian) [inline app/script/view_model/AuthViewModel.js]: Remove empty line

File: app/script/view_model/AuthViewModel.js [hunk 2]
```diff
@@ -0,0 +1,1993 @@
...
+    });
+    this.posted_mode_resend = ko.pureComputed(() => {
+      return this.posted_mode() === z.auth.AuthView.MODE.POSTED_RESEND;
+    });
+    this.posted_mode_retry = ko.pureComputed(() => {
+      return this.posted_mode() === z.auth.AuthView.MODE.POSTED_RETRY;
+    });
+    this.posted_mode_verify = ko.pureComputed(() => {
+      return this.posted_mode() === z.auth.AuthView.MODE.POSTED_VERIFY;
+    });
```
Comment (ffflorian) [inline app/script/view_model/AuthViewModel.js]: Functions in lines 248-262 can all be replaced by the short version:
```js
this.posted_mode_verify = ko.pureComputed(() => this.posted_mode() === z.auth.AuthView.MODE.POSTED_VERIFY);
```

File: app/script/view_model/AuthViewModel.js [hunk 3]
```diff
@@ -0,0 +1,1993 @@
...
+      this.user_service.change_own_password(this.password())
+        .catch((error) => {
+          this.logger.warn('Could not change user password', error);
+          if (error.code !== z.service.BackendClientError.STATUS_CODE.FORBIDDEN) {
+            throw error;
+          }
+        })
+        .then(() => {
+          return this.user_service.change_own_email(this.username());
+        })
```
Comment (ffflorian) [inline app/script/view_model/AuthViewModel.js]: Short function could be applied here, too.

File: app/script/view_model/AuthViewModel.js [hunk 4]
```diff
@@ -0,0 +1,1993 @@
...
+   *
+   * @private
+   * @param {z.auth.AuthView.MODE} mode - View state of the authentication page
+   * @returns {Object} Auth payload for specified mode
+   */
+  _create_payload(mode) {
+    let payload = {};
+    const username = this.username()
+      .trim()
+      .toLowerCase();
```
Comment (ffflorian) [inline app/script/view_model/AuthViewModel.js:584]:
```js
const username = this.username().trim().toLowerCase();
```

File: app/script/view_model/GiphyViewModel.js [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+            this.selected_gif(gifs[0]);
+          }
+          this.state(GiphyViewModel.STATE.RESULTS);
+        })
+        .catch((error) => {
+          this.logger.error(`No gifs found for query: ${this.query()}`, error);
+          this.state(GiphyViewModel.STATE.ERROR);
+        });
+
+    }
```
Comment (ffflorian) [inline app/script/view_model/GiphyViewModel.js]: Remove empty line

File: app/script/view_model/MainViewModel.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+    this.user = this.user_repository.self;
+
+    this.main_classes = ko.pureComputed(() => {
+      if (this.user()) {
+        // deprecated - still used on input control hover
+        let main_css_classes = `main-accent-color-${this.user().accent_id()}`;
+        main_css_classes += ` ${this.user().accent_theme()}`;
+        main_css_classes += ' show';
+        return main_css_classes;
```
Comment (ffflorian) [inline app/script/view_model/MainViewModel.js]:
```js
const main_css_classes = `main-accent-color-${this.user().accent_id()} ${this.user().accent_theme()} show';
```
or
```js
let main_css_classes = `main-accent-color-${this.user().accent_id()} `;
main_css_classes += this.user().accent_theme();
main_css_classes += ' show';
```

File: app/script/view_model/VideoCallingViewModel.js [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+
+    this.remote_video_element_contain = ko.observable(false);
+
+    this.number_of_screen_devices = ko.observable(0);
+    this.number_of_video_devices = ko.observable(0);
+
+    this.calls = this.calling_repository.calls;
+    this.joined_call = this.calling_repository.joined_call;
+
+    this.videod_call = ko.pureComputed(() => {
```
Comment (ffflorian) [inline app/script/view_model/VideoCallingViewModel.js:57]: Shouldn't it be `this.video_call` (everywhere)?

File: app/script/view_model/AuthViewModel.js (hunk 4 above)
Comment (gregor) [reply app/script/view_model/AuthViewModel.js:584]: Linter will fail

File: app/script/view_model/VideoCallingViewModel.js (hunk 7 above)
Comment (gregor) [reply app/script/view_model/VideoCallingViewModel.js:57]: No

File: app/script/view_model/AuthViewModel.js (hunk 4 above)
Comment (lipis) [reply app/script/view_model/AuthViewModel.js:584]: More than two.. :) it's too deep..

**Pull request merged**